lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1263,3 +1271,72 @@ def delete_comment(request, comment_id):
         comment_deleted.send(sender=None, user=request.user, post=cc_comment)
     else:
         raise PermissionDenied
+
+
+@function_trace("_get_course_discussion_user_stats")
+def _get_course_discussion_user_stats(
+    course_key: CourseKey,
+    order_by: UserOrdering = UserOrdering.BY_FLAGS,
+) -> List[Dict[str, Union[str, int]]]:
+    """
+    Get course user discussion stats.
+
+    Args:
+        course_key (CourseKey): course key for course to fetch discussion stats for.
+        order_by (UserOrdering): the ordering to use for sorting data
+
+    Returns:
+        Sorted user stat data for the specified course
+    """
+    course_stats = get_course_user_stats(course_key)
+
+    def key_func(user):
+        key = (user["threads"], user["responses"], user["replies"])
+        if order_by == UserOrdering.BY_FLAGS:
+            key = (user["active_flags"], user["inactive_flags"]) + key
+        return key
+
+    return sorted(course_stats.values(), key=key_func, reverse=True)
+
+
+@function_trace("get_course_discussion_user_stats")
+def get_course_discussion_user_stats(
+    request,
+    course_key_str: str,
+    page: int,
+    page_size: int,
+    order_by: UserOrdering = None,
+) -> Dict:
+    """
+    Get paginated course discussion stats for users in the course.
+
+    Args:
+        request (Request): DRF request
+        course_key_str (str): course key string
+        page (int): Page number to fetch
+        page_size (int): Number of items in each page
+        order_by (UserOrdering): The ordering to use for the user stats
+
+    Returns:
+        Paginated data of a user's discussion stats sorted based on the specified ordering.
+
+    """
+    course_key = CourseKey.from_string(course_key_str)
+    if has_discussion_privileges(user=request.user, course_id=course_key):
+        order_by = order_by or UserOrdering.BY_FLAGS
+    else:
+        order_by = order_by or UserOrdering.BY_ACTIVITY
+        if order_by != UserOrdering.BY_ACTIVITY:
+            raise ValidationError({"order_by": ["Invalid value"]})
+    course_discussion_stats = _get_course_discussion_user_stats(course_key, order_by)
+    paginator = Paginator(course_discussion_stats, page_size)
+    serializer = UserStatsSerializer(paginator.get_page(page), many=True)
+    paginator = DiscussionAPIPagination(
+        request,
+        page,
+        paginator.num_pages,
+        paginator.count,
+    )
+    return paginator.get_paginated_response({
+        "results": serializer.data,
+    })
```

Comment:
@xitij2000 Don't you think this method should only be responsible for providing the data? and pagination/serialization should be part of the API view? I think 1)this function is doing multiple things I) getting course discussion user stats (basic) II) pagination III) serialization of data. 

Do you think it would make this method more reuseable if you move the `paginator` and `serializer` logic in the API get view?

Reply:
If we move the pagination and serializer logic to the view, then there is nothing left here. This function is currently fetching course stats, paginating and serialising that and returning the data. If we move that to the view, then this function will only be setting order_by, in which case it might be better to move everything to the view, and call this method `clean_order_by` and it can simply return a valid ordering. 

Do tell if that's what I should do.

Reply:
I am ok with how it looks today. Thanks for the clarification